framework/src/modules/dpos_v2/commands/pom.ts

```diff
@@ -143,37 +153,53 @@ export class ReportDelegateMisbehaviorCommand extends BaseCommand {
 			getMethodContext(),
 			header1.generatorAddress,
 		);
-		/*
-			Check block signatures validity
-		*/
 		header1.validateSignature(generatorKey, chainID);
 		header2.validateSignature(generatorKey, chainID);
 
-		/*
-			Update sender account
-		*/
 		const delegateAccountBalance = await this._tokenMethod.getAvailableBalance(
 			getMethodContext(),
-			delegateAddress,
+			punishedAddress,
 			this._tokenIDDPoS,
 		);
 
-		const reward =
-			REPORTING_PUNISHMENT_REWARD > delegateAccountBalance
-				? delegateAccountBalance
-				: REPORTING_PUNISHMENT_REWARD;
-
 		delegateAccount.pomHeights.push(currentHeight);
 
-		if (delegateAccount.pomHeights.length >= MAX_POM_HEIGHTS) {
+		this.events.get(DelegatePunishedEvent).log(context, {
+			address: punishedAddress,
+			height: currentHeight,
+		});
+
+		if (delegateAccount.pomHeights.length === POM_LIMIT_BANNED) {
 			delegateAccount.isBanned = true;
+
+			this.events.get(DelegateBannedEvent).log(context, {
+				address: punishedAddress,
+				height: currentHeight,
+			});
 		}
-		await delegateSubStore.set(context, delegateAddress, delegateAccount);
+
+		const currentWeight = getDelegateWeight(
+			BigInt(this._config.factorSelfVotes),
+			delegateAccount.selfVotes,
+			delegateAccount.totalVotesReceived,
+		);
+
+		const eligibleDelegateStore = this.stores.get(EligibleDelegatesStore);
+		await eligibleDelegateStore.update(context, punishedAddress, currentWeight, delegateAccount);
+
+		await delegateSubStore.set(context, punishedAddress, delegateAccount);
+
+		/* Checking if the reward is greater than the delegate's balance. If it is, then it will set the
+		reward to the delegate's balance. */
+		const reward =
+			REPORTING_PUNISHMENT_REWARD > delegateAccountBalance
+				? delegateAccountBalance
+				: REPORTING_PUNISHMENT_REWARD;
 
 		if (reward > BigInt(0)) {
 			await this._tokenMethod.transfer(
 				getMethodContext(),
-				delegateAddress,
+				punishedAddress,
 				transaction.senderAddress,
 				this._tokenIDDPoS,
```

Comment:
token id and amount for reward needs to be different.
Let's create different issue for this